### PR TITLE
Deprecate IActionBlocker in favour of cancellable events

### PR DESCRIPTION
**Pull request by DrSmugleaf**
## About the PR
Implementing 12 methods on an interface just to return false on one of them wasn't that cool.
The old usages still work and are routed through the ActionBlockerSystem, but are marked as obsolete.
Extension methods exist for each message because I'm sure no one wants to write the same 3 lines of code 120 times for each action type.
This PR also moves friction related classes because I put them in a strange place.

### CODE REVIEW

File: Content.Server/Strip/StrippableComponent.cs [hunk 1]
```diff
@@ -29,7 +28,7 @@ public sealed class StrippableComponent : SharedStrippableComponent
         public const float StripDelay = 2f;
 
         [ViewVariables]
-		private BoundUserInterface? UserInterface => Owner.GetUIOrNull(StrippingUiKey.Key);
+        private BoundUserInterface? UserInterface => Owner.GetUIOrNull(StrippingUiKey.Key);
```
Comment (DrSmugleaf) [inline Content.Server/Strip/StrippableComponent.cs:31]: These somehow were 2 tabs before if anyone was wondering.
rider64_lAHpy00GoZ

**Comment by Zumorica** [comment]
This is the PR I've been waiting for my entire life, thank you

**Comment by DrSmugleaf** [comment]
> This is the PR I've been waiting for my entire life, thank you

EffectBlocker's next

**Review by Zumorica — COMMENTED** [review]
Only thing I'm not 100 % about is the extension methods (as Entity System methods should always be used preferably in my opinion) but other than that? Very good

**Comment by DrSmugleaf** [comment]
> Only thing I'm not 100 % about is the extension methods (as Entity System methods should always be used preferably in my opinion) but other than that? Very good
There's no difference to doing it manually other than how annoying it would be, since the semantics of each method are creating and raising an event into the bus and not performing any logic that would belong in a system. Any logic that would prevent the execution of an action would belong in an event listener, those are shorthands for creating/raising/checking the event.

**Comment by Zumorica** [comment]
> There's no difference to doing it manually other than how annoying it would be, since the semantics of each method are creating and raising an event into the bus and not performing any logic that would belong in a system. Any logic that would prevent the execution of an action would belong in an event listener, those are shorthands for creating/raising/checking the event.

I'm not saying we should raise the events manually. In my opinion, organizationally these helper methods belong on Entity Systems. In the future, we will probably remove `IEntity` and `Entity` and putting the methods on entity systems would make the transition much less painful. Extension methods, due to their nature, always feel icky to use. And entity system dependencies will be a thing soon, which would make using these even easier.
I'm not opposing the extension methods, but I'd really prefer using entity system methods

**Comment by DrSmugleaf** [comment]
> In the future, we will probably remove `IEntity` and `Entity` and putting the methods on entity systems would make the transition much less painful.

A job for whenever we start probably coding that I'm sure.


> And entity system dependencies will be a thing soon, which would make using these even easier.

Why would you want an entity system method for raising an event alone though?

File: Content.Shared/DragDrop/DropAttemptEvent.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    {
+        public DropAttemptEvent(IEntity entity)
+        {
+            Entity = entity;
+        }
+
+        public IEntity Entity { get; }
+    }
+
+    public static class DropAttemptExtensions
```
Comment (mirrorcult) [inline Content.Shared/DragDrop/DropAttemptEvent.cs:15]: if you're gonna do these extensions can they really not just be in one ActionBlockerExtensions class?

Comment (Zumorica) [reply Content.Shared/DragDrop/DropAttemptEvent.cs:15]: Don't bother, we discussed this with PJB and agreed to remove the extensions and move them to entity system methods. I volunteered to change it.

Comment (Zumorica) [reply Content.Shared/DragDrop/DropAttemptEvent.cs]: done

**Pull request merged**